### PR TITLE
fix: [CDS-82034]: ServiceNow create/update :when ticket type is reset or changes all fields should be cleared out and reducing excess metaData calls in jira update/create

### PR DESCRIPTION
**Pull request by satyam-harness**
Problems - 

1. when ticket type is reset or changes all fields should be cleared out. (https://harness.atlassian.net/browse/CDS-82034)
2. display value was getting saved in yaml instead of value for snow update fields. (https://harness.atlassian.net/browse/CDS-81993)
3. reducing excess metaData calls in jira create/update. (https://harness.atlassian.net/browse/CDS-77629, https://harness.atlassian.net/browse/CDS-77630)

Solution - 
1. cleared all dependent fields when ticketType is changed
2. used value instead of displayValue
3. removing metaData api calls from `JiraDynamicFieldSelector` and reusing calls made in jira steps.

Manual testing - 

1. Tested issueMetaData and projectMetaData are getting passed properly.
2. Tested refecthIssueMetaData and refetchProjectMetadata are getting called upon changing project/issueType
3. Dependent fields are getting cleared upon changing ticketType
4. value is getting saved in all case instead of display value 

https://github.com/harness/harness-core-ui/assets/96036463/8b81c703-f180-408e-b82a-a3e6daa71a6d

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger eslint

**Comment by satyam-harness** [comment]
retrigger eslint

**Comment by satyam-harness** [comment]
retrigger eslint

**Comment by satyam-harness** [comment]
trigger messagecheck

**Review by sahithikolichala — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]

**Review by Eric761 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraCreateStepMode.tsx [hunk 1]
```diff
@@ -328,6 +330,14 @@ function FormContent({
         title={getString('pipeline.jiraCreateStep.addFields')}
       >
         <JiraDynamicFieldsSelector
+          refetchProjectMetadata={refetchProjectMetadata}
+          refetchIssueMetadata={refetchIssueMetadata}
+          fetchingIssueMetadata={fetchingIssueMetadata}
+          fetchingProjectMetadata={fetchingProjectMetadata}
+          projectMetaResponse={projectMetaResponse}
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraCreateStepMode.tsx:337]: nit: projectMetadataResponse

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraCreateStepMode.tsx [hunk 2]
```diff
@@ -328,6 +330,14 @@ function FormContent({
...
+          issueMetaResponse={issueMetaResponse}
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraCreateStepMode.tsx:338]: Same as above

**Review by chetanvekariya-harness — APPROVED** [review]
Plan for test cases?

**Comment by satyam-harness** [comment]
trigger messagecheck

**Pull request merged**